core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -136,7 +136,7 @@ protected long targetSplitSize(TableOperations ops) {
         } else {
           return StaticDataTask.of(
               ops.io().newInputFile(ops.current().file().location()),
-              snap.manifests(),
+              snap.dataManifests(), // if the manifest list is missing, the table must be v1 and has no delete manifests
```

Comment:
Shouldn't this table return all manifests?

Reply:
This else block is taken when manifests aren't kept in a file and are embedded in table metadata. That's not allowed in v2, so delete manifests and this code path will not happen at the same time.

We could be more explicit and use `allManifests`, so that this fails if a delete manifest is found. That's probably easier to understand and safer.

Reply:
Updated to use `allManifests`.